fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -373,7 +378,7 @@ public <R> CompletableFuture<R> applyInRecoveryPath(Function<FDBRecordContext, C
                 future = function.apply(recoveryContext)
                         .whenComplete((result, ex) -> {
                             if (ex == null) {
-                                recoveryContext.commit();
+                                commit(recoveryContext);
```

Comment:
After this PR, at some point....
This was in existing code, but this probably shouldn't be calling `asyncToSync`, and instead should be doing
```
function.apply(recoveryContext)
    .thenCompose(recoveryContext.commitAsync)
    .whenComplete(() -> recoveryContext.close());
```

Reply:
Added to future issues.
Together with `LucenePartitioner`:
```
public PartitionedQueryHint selectQueryPartition(@Nonnull Tuple groupKey, @Nullable LuceneScanQuery luceneScanQuery) {
        return LuceneConcurrency.asyncToSync(WAIT_LOAD_LUCENE_PARTITION_METADATA, selectQueryPartitionAsync(groupKey, luceneScanQuery), state.context);
}
```